orderbook/src/main.rs

```diff
@@ -151,6 +151,19 @@ async fn main() {
         .expect("Could not get chainId")
         .as_u64();
 
+    let vault = match chain_id {
+        1 | 4 => Some(
+            BalancerV2Vault::deployed(&web3)
```

Comment:
Would be slightly cleaner to inspect the deployments that ethcontract knows about directly and compare it to the chain id instead of first matching on chain id and then calling deployed which internally retrieves the chain id again. That way it is clear that we always look at the same chain id and automatically handle the case where balancer adds a new deployment on a new chain id without touching this.
Not sure if this possible with the current ethcontract api, just mentioning.

Reply:
Changed this to retrieve the network and check whether or not it is part of the deployment.